systers_portal/meetup/views.py

```diff
@@ -123,9 +125,16 @@ def get_redirect_url(self, *args, **kwargs):
         new_meetup.slug = meetup_request.slug
         new_meetup.date = meetup_request.date
         new_meetup.time = meetup_request.time
-        new_meetup.venue = meetup_request.venue
         new_meetup.description = meetup_request.description
-        new_meetup.meetup_location = meetup_request.meetup_location
+        if not meetup_request.is_virtual:
+            new_meetup.venue = meetup_request.venue
+            new_meetup.meetup_location = meetup_request.meetup_location
+        if meetup_request.is_virtual:
+            meet_data = create_meetup(new_meetup)
+            new_meetup.meet_link = meet_data['join_url']
```

Comment:
For places where you have directly accessed the `dict`, I would check for `KeyError`, since you are directly using the data from Zoom.

If it's KeyError, send back an HTTP 500 or something.

Reply:
Done